test/cases/compile_errors/invalid_type_in_builtin_extern.zig

```diff
@@ -0,0 +1,11 @@
+const x = @extern(*u3, .{.name="foo"});
+pub export fn entry() void {
+    _ = x;
+}
+
+// error
+// backend=stage2
+// target=native
+//
+// :1:19: error: extern symbol cannot have type '*u3'
+// :1:19: note: only integers with power of two bits are extern compatible
```

Comment:
Maybe we can check comptime types too? Here's formatting and another test case:
```suggestion
export fn entry1() void {
    const x = @extern(*u3, .{ .name = "foo1" });
    _ = x;
}
export fn entry2() void {
    const x = @extern(*comptime_int, .{ .name = "foo2" });
    _ = x;
}

// error
// backend=stage2
// target=native
//
// :1:19: error: extern symbol cannot have type '*u3'
// :1:19: note: only integers with power of two bits are extern compatible
// :2:19: error: extern symbol cannot have type '*comptime_int'
```
(check this though, not sure if this is right as-is)